### ISSUE TITLE
Manage content types

### ISSUE THREAD

**Issue by DariusIMP** [DariusIMP/zenoh-backend-s3#2]
So far, for development purposes (as it's easier to test), the preliminary version of the S3 Plugin treats all the objects retrieved from s3 storage as strings. This is probably not going to be the case for many use cases, so we need to be able to distinguish which kind of encoding uses a stored object. 
Probably the way to go is to store some user defined metadata fields upon uploading the files and later retrieve them during the query, decoding the file accordingly.

**Comment by clibequilibrium** [comment]
Hello @DariusIMP 

Will it make more sense to use S3 provided API to get MIME type of the object?
https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadObject.html

Minio also supports custom mime types for some file extensions so the metadata is already provided.

![image](https://user-images.githubusercontent.com/105135724/194619607-adbd71b1-48fd-4946-8cad-1b472cb9664b.png)

**Comment by DariusIMP** [comment]
Yep, it makes more sense and I've been already working it. Thankfully the aws-sdk-s3 provides a [simple method](https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/client/fluent_builders/struct.CopyObject.html#method.set_content_encoding) to solve this problem.
